tools/setup/emoji/generate_emoji_names

```diff
@@ -69,6 +69,23 @@ SKIN_TONE_EMOJI_CODES = [
 ]
 
 
+def cleanup_name(name: str) -> str:
+    replacements = {
+        " ": "_",
+        "“": "",
+        "”": "",
+        ":": "",
+        ".": "",
+        "&": "and",
+        "‘": "'",
+        "’": "'",
+        "–": "-",
+    }
+    for before, after in replacements.items():
+        name = name.replace(before, after)
+    return name.lower()
```

Comment:
I think this algorithm might be incomplete; this commit leaves us with a `flag R\u00e9union` canonical name.

Reply:
This was handled in the custom emoji list [here](https://github.com/zulip/zulip/pull/22067/files/7a18bbe555e2598e9cffa847f52914b3574b2ece#diff-bdbbd354daa882573f5d612653e8e18dc4791a25723fe1fa58fdc7dd313aac18R1595). (I forgot to put `_` there, which signals to me that it would be good to run `cleanup` on the custom names as well as the CLDR names just in case. )

I kept the correct-in-that-language spelling as the canonical name in situations like that (since I felt like that was respectful to that language), and added aliases for "English"-letter spellings so that the emoji could still be accessible without having to type those characters. I can add a comment explaining that thinking. 

I hadn't generalized that because there are a lot of non-standard characters, but I just added `convert_non_ascii_chars` and a block of code that uses that. Let me know what you think :)

Reply:
I think this is a reasonable policy -- but this sort of decision needs to be documented with comments :)

Reply:
I had used a short comment elsewhere in the code that was a TODO describing that I wasn't doing this yet. Agreed that I could have added more context about why I didn't, and maybe also made a comment in the `cleanup` function. But there definitely were comments about it.

Now there is a function that handles this, so it feels like I don't need to comment anymore, especially because an error will be raised if any new non-ascii characters are added in the future.